src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs

```diff
@@ -49,8 +50,21 @@ internal class DefaultRazorFormattingService : RazorFormattingService
                 result = await pass.ExecuteAsync(context, result, cancellationToken);
             }
 
-            var filteredEdits = result.Edits.Where(e => range.LineOverlapsWith(e.Range)).ToArray();
-            return filteredEdits;
+            var filteredEdits = result.Edits.Where(e => range.LineOverlapsWith(e.Range));
+
+            // Make sure the edits actually change something, or its not worth responding
+            var textChanges = filteredEdits.Select(e => e.AsTextChange(originalText));
+            var changedText = originalText.WithChanges(textChanges);
+            if (changedText.ContentEquals(originalText))
+            {
+                return Array.Empty<TextEdit>();
+            }
+
+            // Only send back the minimum edits
+            var minimalChanges = SourceTextDiffer.GetMinimalTextChanges(originalText, changedText, lineDiffOnly: false);
```

Comment:
Ah I see, before we were only doing the minimal changes in the pass itself not at the top-level formatting endpoint